lib/Dialect/Transforms/IsolateGroupOpsPass.cpp

```diff
@@ -100,7 +112,8 @@ class TcpIsolateGroupOpsPass
     MLIRContext *context = op->getContext();
     RewritePatternSet patterns(context);
 
-    patterns.add<IsolateGroups>(context);
+    auto shouldCopyConstPredicate = [&](tcp::GroupOp, Value) { return true; };
+    populateIsolateGroupPatterns(patterns, shouldCopyConstPredicate);
```

Comment:
Couple of questions:
1. Whats the adv of having a lambda over a bool to control this?
2. Why are we passing this to `populateIsolateGroupPatterns` instead of adding it as an option to the pass itself?

Reply:
Recapping offline discussion:
1. The downstream uses of this API do have more complicated logic to figure out when to inline a const (for instance based on users of that const, compilation mode etc.)
2. I am not sure there's an easy way to add a boolean predicate function as an option to a pass itself in the .td file. I'd be happy to also add a default functor argument to the pass in that case.